keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -1066,6 +1076,264 @@ the remaining requests in that queue start getting faster service.  In
 both cases, the service delivery in the virtual world has reacted
 properly to the true service duration.
 
+### Support for LIST requests
+
+Up until now, we were assuming that even though the requests aren't
+necessarily equally expensive, their actual cost is actually greatly
+reflected by the time it took to process them.  But while being processed
+each of them is consuming the equal amount of resources.
+
+It works well for requests that are touching only a single object.
+However, given the fact that in practise the concurrency limits has to be
+set much higher than number of available cores to achieve reasonable system
+throughput, this no longer works that well for LIST requests that are orders
+of magnitude more expensive.  The hand-wavy way of rationalizing it is that
+the ratio of time the request is processed by the processor to the actual
+time of processing the request starts to visibly differ (e.g. due to I/O
+waiting time - there is communication with etcd in between for example).
+
+As a result, kube-apiserver (and etcd) may be able to easily keep with N
+simple in-flight requests (e.g. create or get a single Pod), but will explode
+trying to process N requests listing all the pods in the system at the same
+time.
+
+#### Width of the request
+
+In order to address this problem, we are introducing the concept of `width`
+of the request.  Instead of saying that every request is consuming a single
+unit of concurrency, we allow for a request to consume `<width>` units of
+concurrency while being processed.
+
+While in theory the `width` can be an arbitrary non-integer number, for
```

Comment:
I think of `width` as a multiplying factor to the `estimated service time` of a request in the virtual world. In the future we can extend `width` of a request and calculate it from different real world sources like cpu/memory usage or some etcd stress indicator and feed it into the virtual world. Just a thought.